src/aspire/source/image.py

```diff
@@ -345,13 +345,17 @@ def _images(self, start=0, num=np.inf, indices=None):
             "Subclasses should implement this and return an Image object"
         )
 
-    def _apply_filters(self, im_orig, start=0, num=np.inf, indices=None):
+    def _apply_filters(
```

Comment:
This signature can now be simplified, since `start`, `num`, and `indices` don't really fill a function anymore. It's just enough to pass `filters` and `filter_indices` to `_apply_filters`. Then `_apply_src_filters` and `_apply_sim_filters` can deal indexing by passing `filter_indices[indices]`. The `start` and `num` parameters can go away completely here (just create the `np.arange` in `im_backward` and `vol_forward`.

Reply:
This change triggers the other bug in #549! Maybe given how close I keep coming to hitting this bug, I should go fix that one before we merge this in.

Reply:
Will move #549 to higher priority